.github/workflows/publish.yml

```diff
@@ -0,0 +1,66 @@
+name: Publish artifacts
+
+on:
+  release:
+    ### NOTE:
+    ###
+    ### There are multiple types of release trigger events.
+    ### Ref: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release
+    ###
+    ### Given that GitHub allows the creation of 'release'
+    ### and 'pre-release' publications to differentiate between
+    ### production and beta releases, the trigger type to use,
+    ### as it wraps them both, must be 'published'.
+    types: [published]
+
+jobs:
+
+  docker:
+    needs: []
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Login to DockerHub"
```

Comment:
Maybe also deploy to GitHub Container Registry? c.f. https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/.github/workflows/docker.yml#L61-L67

Reply:
I believe no one uses the GitHub Container Registry for this project, but @johannbrehmer , @KlingFelix  or @irinaespejo may know better 👀

Reply:
Sure. I'm starting to use it just as an alternative to having things only be on Docker Hub. https://github.com/orgs/scikit-hep/packages/container/package/pyhf

Definitely totally optional. :+1: